scripts/Muon/GUI/ElementalAnalysis2/context/data_context.py

```diff
@@ -0,0 +1,89 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from typing import List
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+import Muon.GUI.Common.utilities.load_utils as load_utils
+
+
+class DataContext(object):
+    def __init__(self, load_data=MuonLoadData()):
+        self.instrument = "rooth"
+        self.current_runs = []
+        self._loaded_data = load_data
+        self._run_info = []
+        self.previous_runs = []
+
+    @property
+    def run_info(self):
+        return self._run_info
+
+    def run_info_update(self, run_object):
+        self._run_info.append(run_object)
+
+    def clear_run_info(self):
+        self._run_info = []
+
+    @property
+    def num_detectors(self):
+        try:
+            n_det = self.current_workspace.detectorInfo().size()
+        except AttributeError:
+            # default to 1
+            n_det = 1
+        if n_det == 0:
+            # The number of histograms commonly used for PSI data, in real data,
+            # very often the number of detectors == number of histograms
+            return self.current_workspace.getNumberHistograms()
```

Comment:
this wont work here. We have 3 histograms in a workspace, but from the same detector.